File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/spelling/SpellingSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.spelling;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.nio.file.Path;
import java.util.List;

@Value.Immutable
@JsonDeserialize(as = ImmutableSpellingSettings.class)
public interface SpellingSettings {
  List<Path> dictionaries();
}

File: layout/src/main/kotlin/com/github/alexandrecarlton/idea/settings/dialog/editor/spelling/SpellingSettings.kt
package com.github.alexandrecarlton.idea.settings.dialog.editor.spelling

import com.fasterxml.jackson.annotation.JsonProperty
import java.io.File

data class SpellingSettings(
    @JsonProperty("Dictionaries")
    val dictionaries: List<File>? = null,

    @JsonProperty("Accepted Words")
    val acceptedWords: List<String>? = null
)
